src/routing/approved.rs

```diff
@@ -1996,8 +2103,19 @@ impl Approved {
 
         trace!("relay {:?} to {:?}", msg, targets);
 
-        let targets: Vec<_> = targets.into_iter().map(|node| *node.addr()).collect();
-        let command = Command::send_message_to_nodes(&targets, dg_size, msg.to_bytes());
+        let targets: Vec<_> = targets
+            .into_iter()
+            .map(|node| (*node.addr(), *node.name()))
+            .collect();
+        let command = Command::send_message_to_nodes(
+            targets,
+            dg_size,
+            msg.to_bytes(),
+            HeaderInfo {
+                dest: XorName::random(),
```

Comment:
if it always involve using a random input then later on replacing, 
shall we consider to make the field option?
filling with `random` does looks quite suspicious

Reply:
Yea, makes sense. Makes the code cleaner as well.